HandEyeCalibration/Transform/Transform.cpp

```diff
@@ -0,0 +1,343 @@
+        Eigen::AngleAxisd(representations.rotationVector.norm(), representations.rotationVector.normalized());
+    std::cout << "\nAxis-angle:\n" << reorderAxisAngle(axisAngle) << std::endl;
+
+    // Axis-Angle to Quaternion
+    std::cout << "\nConverting axis-angle to quaternion" << std::endl;
+    const Eigen::Quaterniond quaternion = Eigen::Quaterniond(axisAngle);
+    std::cout << "\nQuaternion:\n" << quaternion.coeffs() << std::endl;
+
+    // Quaternion to Rotation Matrix
+    std::cout << "\nConverting quaternion to rotation matrix" << std::endl;
+    const Eigen::Matrix3d rotationMatrixFromQuaternion = quaternion.toRotationMatrix();
+    std::cout << "\nRotation Matrix:\n" << rotationMatrixFromQuaternion << std::endl;
+    fileStorageOut.write("RotationMatrixFromQuaternion", eigenToCv(rotationMatrixFromQuaternion));
+
+    // Calculate error in transformation (Rotation Matrix --> Quaternion --> Rotation Matrix)
+    std::cout << "\nVerifying Rotation Matrix from Quaternion" << std::endl;
+    compareRotation(rotationMatrix, rotationMatrixFromQuaternion);
+
+    fileStorageOut.release();
+}
+
+Eigen::Vector4d reorderAxisAngle(const Eigen::AngleAxisd &axisAngle)
+{
+    Eigen::Vector4d reorderedAxisAngle;
+
+    for(int i = 0; i < 3; i++)
+        reorderedAxisAngle[i] = axisAngle.axis()[i];
+    reorderedAxisAngle[3] = axisAngle.angle();
+
+    return reorderedAxisAngle;
+}
+
+/** Converts Rotation Matrix to RollPitchYaw.
+    * The convention we use here is that Roll is a rotation about x-axis,
+    * Pitch is a rotation about y-axis and Yaw is a rotation about z-axis.
+    * @param rotationMatrix Rotation Matrix to convert to RPY
+    * @param rotation Enum to indicate which rotation convention to follow for this conversion
+    * @return An array of angles, in radians
+    */
+Eigen::Array3d rotmat2rpy(const Eigen::Matrix3d &rotationMatrix,
+                          const RotationConvention &rotation = RotationConvention::ZYX_Intrinsic)
+{
+    Eigen::Array3d rollPitchYaw;
+    switch(rotation)
+    {
+        case RotationConvention::XYZ_Intrinsic: rollPitchYaw = rotationMatrix.eulerAngles(0, 1, 2); break;
+        case RotationConvention::XYZ_Extrinsic: rollPitchYaw = rotationMatrix.eulerAngles(2, 1, 0); break;
+        case RotationConvention::ZYX_Intrinsic: rollPitchYaw = rotationMatrix.eulerAngles(2, 1, 0); break;
+        case RotationConvention::ZYX_Extrinsic: rollPitchYaw = rotationMatrix.eulerAngles(0, 1, 2); break;
+        default: throw std::invalid_argument("Invalid rotation");
+    }
+    // The returned array should be ordered by Ox, Oy and Oz, where Oa is the rotation about axis a.
+    switch(rotation)
+    {
+        case RotationConvention::XYZ_Extrinsic: rollPitchYaw.reverseInPlace(); break;
+        case RotationConvention::ZYX_Intrinsic: rollPitchYaw.reverseInPlace(); break;
+    }
+
+    return rollPitchYaw;
+}
+
+/** Converts RollPitchYaw (in radians) to Rotation Matrix.
```

Comment:
`RollPitchYaw` is a specific type in your program, which this function doesn't take as a parameter. Maybe you can write this comment a bit differently to avoid the confusion.